pulser-core/pulser/channels/eom.py

```diff
@@ -0,0 +1,170 @@
+
+    mod_bandwidth: float  # MHz
+
+    def __post_init__(self) -> None:
+        if self.mod_bandwidth <= 0.0:
+            raise ValueError(
+                "'mod_bandwidth' must be greater than zero, not"
+                f" {self.mod_bandwidth}."
+            )
+
+    @property
+    def rise_time(self) -> int:
+        """The rise time (in ns).
+
+        Defined as the time taken to go from 10% to 90% output in response to
+        a step change in the input.
+        """
+        return int(MODBW_TO_TR / self.mod_bandwidth * 1e3)
+
+    def _to_dict(self) -> dict[str, Any]:
+        params = {
+            f.name: getattr(self, f.name) for f in fields(self) if f.init
+        }
+        return obj_to_dict(self, **params)
+
+
+@dataclass(frozen=True)
+class RydbergEOM(BaseEOM):
+    """The EOM configuration for a Rydberg channel.
+
+    Attributes:
+        mod_bandwidth: The EOM modulation bandwidth at -3dB (50% reduction),
+            in MHz.
+        limiting_beam: The beam with the smallest amplitude range.
+        max_limiting_beam: The maximum amplitude the limiting beam can reach,
+            in rad/µs.
+        intermediate_detuning: The detuning between the two beams, in rad/µs.
+        controlled_beams: The beams that can be switched on/off with an EOM.
+    """
+
+    limiting_beam: RydbergBeam
+    max_limiting_amp: float  # rad/µs
+    intermediate_detuning: float  # rad/µs
+    controlled_beams: tuple[RydbergBeam, ...]
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        for param in ["max_limiting_amp", "intermediate_detuning"]:
+            value = getattr(self, param)
+            if value <= 0.0:
+                raise ValueError(
+                    f"'{param}' must be greater than zero, not {value}."
+                )
+        if not isinstance(self.controlled_beams, tuple):
+            if not isinstance(self.controlled_beams, list):
+                raise TypeError(
+                    "The 'controlled_beams' must be provided as a tuple "
+                    "or list."
+                )
+            # Convert list to tuple to keep RydbergEOM hashable
+            object.__setattr__(
+                self, "controlled_beams", tuple(self.controlled_beams)
+            )
+        if not self.controlled_beams:
+            raise ValueError(
+                "There must be at least one beam in 'controlled_beams'."
+            )
+        for beam in chain((self.limiting_beam,), self.controlled_beams):
+            if not (isinstance(beam, RydbergBeam) and beam in RydbergBeam):
+                raise TypeError(
+                    "Every beam must be one of options of the `RydbergBeam`"
+                    f" enumeration, not {self.limiting_beam}."
+                )
+
+    def detuning_off_options(
+        self, rabi_frequency: float, detuning_on: float
+    ) -> list[float]:
+        """Calculates the possible detuning values when the amplitude is off.
```

Comment:
This is the detuning that is applied because of the presence of the EOM during free evolution, right?